scripts/apps/authoring/authoring/authoring-custom-field.tsx

```diff
@@ -70,13 +72,26 @@ export class AuthoringCustomField extends React.PureComponent<IProps, IState> {
 
         return (
             <div>
-                <FieldType.editorComponent
-                    item={item}
-                    value={this.state.value}
-                    setValue={(value) => this.setValue(value)}
-                    readOnly={!editable}
-                    config={field.custom_field_config}
-                />
+                {this.props.template == null ?
```

Comment:
Shouldn't it determine which component to render, depending on whether `templateEditorComponent` exists? Not based on the current value of `template`?

Reply:
we have defined the `templateEditorComponent` in customFieldTypes [here](https://github.com/superdesk/superdesk-client-core/blob/32395021e0da0fe595d2248a60ada727ec232d18/scripts/extensions/datetimeField/src/extension.tsx#L50), so don't you think it's gonna be present every time?

Reply:
You have marked it [optional](https://github.com/superdesk/superdesk-client-core/pull/3864/files#diff-ca49bae7f62187bd0909482a419226feafbf06df59d2d06c2031cd298e7d62aaR1845) in the interface. And that's a good thing. In most cases, when fields are static, using `editorComponent` will work fine when configuring a template.

`templateEditorComponent` would be needed for advanced functionality like dynamic templates, but when an extension doesn't need that advanced functionality, we shouldn't be making it harder for the extension author by requiring it.